aten/src/ATen/Context.h

```diff
@@ -114,12 +114,42 @@ class CAFFE2_API Context {
   void setDeterministicCuDNN(bool);
   bool deterministic() const;
   void setDeterministic(bool);
+
+  // Throws an error if `Context::deterministic()` is true. This method should be called by
+  // operations that act nondeterministically. Most of the time, this method should be called
+  // like this:
+  //  at::globalContext().alertNotDeterministic("<name of operation>");
+  //
+  // All operations that call this method must be documented in torch/__init__.py:set_deterministic()
+  //
+  // All callsites should have a comment explaining why the operation is nondeterministic.
+  //
+  // To help enforce full documentation, all callsites should also have the following note:
+  // NOTE: If you copy and paste this alert to a different operation, you must add a
+  //    comment explaining why the operation is nondeterministic, document the operation
+  //    in torch/__init__.py:set_deterministic(), and copy and paste this note.
   void alertNotDeterministic(c10::string_view const& caller);
+
   bool allowTF32CuDNN() const;
   void setAllowTF32CuDNN(bool);
   bool allowTF32CuBLAS() const;
   void setAllowTF32CuBLAS(bool);
+
+  // Throws an error if `Context::deterministic()` is true and CUDA >= 10.2 and
+  // CUBLAS_WORKSPACE_CONFIG is not set deterministically. This method should be called
```

Comment:
What does it mean for CUBLAS_WORKSPACE_CONFIG to "[not be] set deterministically"?

Reply:
I've changed this to describe exactly which two settings won't cause an error.